### DISCUSSION TITLE
How to lock the first element so that it is an h1 element?

### DISCUSSION THREAD

**Discussion by brunowego** [prosekit/prosekit#597] [Q&A]
Exactly how Notion is doing:

<img width="639" alt="Screenshot 2024-06-28 at 4 55 36 PM" src="https://github.com/prosekit/prosekit/assets/441774/52e60d2f-9ad9-474f-bd16-200413e3b87f">

Thanks!

**Comment by ocavue** [comment]
You can use `defineNodeSpec()` to define a `doc` node with `content: 'heading block+'`

```ts
function defineCustomDoc() {
  return defineNodeSpec({
    name: 'doc',
    content: 'heading block+',
    topNode: true,
  })
}
```

Here is a full example https://codesandbox.io/p/devbox/stoic-wing-5dfflg
